internal/api/mfa.go

```diff
@@ -61,6 +85,168 @@ const (
 	QRCodeGenerationErrorMessage   = "Error generating QR Code"
 )
 
+func initializeWebAuthnConfig(config *WebAuthnConfiguration) (*webauthn.WebAuthn, error) {
+	if config == nil {
+		return nil, badRequestError(ErrorCodeValidationFailed, "WebAuthn config required")
+	}
+	if config.RPDisplayName == "" {
+		return nil, badRequestError(ErrorCodeValidationFailed, "WebAuthn Display name cannot be empty")
+	}
+	if config.RPID == "" {
+		return nil, badRequestError(ErrorCodeValidationFailed, "WebAuthn RP ID cannot be empty")
+	}
+	if len(config.RPOrigins) == 0 {
+		return nil, badRequestError(ErrorCodeValidationFailed, "WebAuthn RP Origins cannot be empty")
+	}
+
+	var invalidOrigins []string
+
+	for _, origin := range config.RPOrigins {
+		parsedURL, err := url.Parse(origin)
+		if err != nil || (parsedURL.Scheme != "https" && parsedURL.Scheme != "http") || parsedURL.Host == "" {
+			invalidOrigins = append(invalidOrigins, origin)
+		}
+	}
+	if len(invalidOrigins) > 0 {
+		return nil, badRequestError(ErrorCodeValidationFailed, fmt.Sprintf("Invalid RP origins found: %v", invalidOrigins))
+	}
+	wconfig := &webauthn.Config{
+		RPDisplayName: config.RPDisplayName,
+		RPID:          config.RPID,
+		RPOrigins:     config.RPOrigins,
+	}
+	webAuthn, err := webauthn.New(wconfig)
+	if err != nil {
+		return nil, badRequestError(ErrorCodeValidationFailed, fmt.Sprintf("invalid webauthn configuration: %v", err))
```

Comment:
Can you please be careful with the casing? Either use `webauthn` or `WebAuthn` everywhere.